src/util/chooseRenormalize.ts

```diff
@@ -0,0 +1,64 @@
+function mean(values: number[]): number {
+  return values.reduce((memo: number, i: number) => memo + i) / values.length;
+}
+
+/**
+ * This defines a random number between 0 and "mass", which should cover the entire
+ * domain of the probability distribution. Then we will "walk" the probability mass
+ * until we exceed the generated value
+ */
+function basicChooseOne(probs: number[], mass: number): number {
+  const r = mass * Math.random();
+  let i = 0;
+  let s = probs[0];
+  while (s < r) {
+    i += 1;
+    s += probs[i];
+  }
+  return i;
+}
+
+function chooseRenormalize(probs: number[], k: number): number[] {
+  //  If we have more work slots than candidates in the probability distribution,
+  //  we return them all
+  if (k >= probs.length)
+    return Array(probs.length)
+      .fill(null)
```

Comment:
It would be good to see the size of the core-js `Array.prototype.fill()` polyfill that babel preset env compiles for this, as it it may not be worth the overhead.